components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -553,6 +603,31 @@ Some claims (Eg:- Birth date) can have "/" in claim values. But in user store le
         }
     }
 
+    /**
+     * Get all the domain names related to user stores.
+     * @param tenantId   Tenant ID
+     * @return A list of all the available domain names for given tenant
+     */
+    private String[] getDomainNames(int tenantId) throws IdentityRecoveryServerException {
+
+        List<String> domainsOfUserStores = new ArrayList<>();
+        AbstractUserStoreManager carbonUM = (AbstractUserStoreManager) getUserStoreManager(tenantId);
+        UserStoreManager secondaryUserStore = carbonUM.getSecondaryUserStoreManager();
+        while (secondaryUserStore != null) {
+            String domainName = secondaryUserStore.getRealmConfiguration().
+                    getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_DOMAIN_NAME).toUpperCase();
+            secondaryUserStore = secondaryUserStore.getSecondaryUserStoreManager();
+            domainsOfUserStores.add(domainName);
+        }
+        // Sorting the secondary user stores to maintain an order fo domains so that pagination is consistent.
+        Collections.sort(domainsOfUserStores);
```

Comment:
Why do we sort here?
What criteria is used to sort?

Reply:
We do not need that sorting, will remove this.

Reply:
Resolved by https://github.com/wso2-extensions/identity-governance/pull/623/commits/00a46fd2ecefd9f65eeebbacfa8e641f455a6a60